### PR TITLE
Feat/export sql script

### PR DESCRIPTION
**Pull request by buddiman**
## Proposed features
This is a first draft how to implement https://github.com/azerothcore/Keira3/issues/401  

I created a rudimentary export functionality, I created a PR already as I'm open to more input/suggestions/requirements.
At the moment, this only works for the *creature_template* editor.

To test it, enter a path to a (non)existing SQL file in the new textbox in the dashboard:  
image
On Unix systems you have to manually edit the path in the textbox for now as I used the `\` character

I created a new service for handling the functionality as it may can be used later for other features.

## Ideas
- Implement https://github.com/azerothcore/Keira3/issues/741
- Directly run the bash script from AC to create a new SQL update file 
- Maybe this can be used for https://github.com/azerothcore/Keira3/issues/1086

### CODE REVIEW

File: src/app/shared/modules/query-output/query-output.component.ts [hunk 1]
```diff
@@ -46,6 +47,14 @@ export class QueryOutputComponent<T extends TableRow> extends SubscriptionHandle
     }
   }
 
+  saveToSQL(): void {
+    if (this.showFullQuery()) {
+      this.saveToSQLFile.emit(this.editorService.fullQuery);
+    } else {
+      this.saveToSQLFile.emit(this.editorService.diffQuery);
+    }
```
Comment (Helias) [inline src/app/shared/modules/query-output/query-output.component.ts:55]:
```suggestion
    const fullOrDiffQuery = this.showFullQuery() ? this.editorService.fullQuery : this.editorService.diffQuery);
    this.saveToSQLFile.emit(fullOrDiffQuery);
```

File: src/app/shared/services/save-query.service.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+export class SaveQueryService {
+  filePath: string = '';
+
+  appendToFile(query: string) {
+    fs.appendFile(this.filePath, query + '\n', { flag: 'a+' }, (err) => {
+      if (err) console.log(err);
+    });
+  }
+
+  setFilePath(path: string) {
```
Comment (Helias) [inline src/app/shared/services/save-query.service.ts:16]:
```suggestion
  setFilePath(path: string): void {
```

File: src/app/shared/services/save-query.service.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import * as fs from 'fs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SaveQueryService {
+  filePath: string = '';
```
Comment (Helias) [inline src/app/shared/services/save-query.service.ts:8]: if the type is inferred, we usually avoid writing it explicitly
```suggestion
  filePath = '';
```

File: src/app/shared/services/save-query.service.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+  appendToFile(query: string) {
+    fs.appendFile(this.filePath, query + '\n', { flag: 'a+' }, (err) => {
+      if (err) console.log(err);
```
Comment (Helias) [inline src/app/shared/services/save-query.service.ts:12]:
```suggestion
      if (err) {
        console.log(err);
      }
```

**Comment by Helias** [comment]
Thanks for your PR, I don't know if you will take care of the unit-test of your service but they are really important to do for the Keira3 project.

Moreover, is the path can be chosen when the user click on "save to sql"? and is there a way to easily choose a path and make it as default without writing it?

**Comment by buddiman** [comment]
> Thanks for your PR, I don't know if you will take care of the unit test of your service but they are really important to do for the Keira3 project.

It's basically a WIP at the moment, I only already opened the PR to get early feedback on what is needed/wished.
I'll do unit tests for sure.

> Moreover, is the path can be chosen when the user clicks on "save to SQL"? and is there a way to easily choose a path and make it as default without writing it?
I want to change the textbox in the dashboard so that you can open a "file explorer dialog" as electron already provides this feature. As for having a default path, that would be a good idea.

**Comment by FrancescoBorzi** [comment]
closing as it looks like abandoned

**Pull request closed without merging**